env/env_posix.cc

```diff
@@ -528,4 +532,12 @@ std::unique_ptr<Env> NewCompositeEnv(std::shared_ptr<FileSystem> fs) {
   return std::unique_ptr<Env>(new PosixEnv(default_env, fs));
 }
 
+std::unique_ptr<Env> NewCompositeEnv(std::shared_ptr<FileSystem> fs,
```

Comment:
This seems off to me.  Why are we potentially creating two Envs in order to create/return one?

Reply:
The idea is that we create a base env that can be shared among composite envs, and that we can control the order of destruction of the base env and its dependent composite envs in a separate c-file.